src/VtkGeometryRenderer.js

```diff
@@ -0,0 +1,112 @@
+
+  getDefaultProps() {
+    return {
+      className: '',
+      showFPS: false,
+      style: {},
+      viewId: -1,
+      interactionTimout: 500,
+    };
+  },
+
+  componentWillMount() {
+    // Make sure we monitor window size if it is not already the case
+    console.log('Mounting VtkGeometryRenderer');
+    // sizeHelper.startListening();
+  },
+
+  componentDidMount() {
+    console.log('VtkGeometryRenderer is now mounted');
+
+    // Get our hands on the default synchronization context and tell how to fetch data arrays
+    const synchCtx = vtkSynchronizableRenderWindow.getSynchronizerContext(SYNCHRONIZATION_CONTEXT_NAME);
+    synchCtx.setFetchArrayFunction(this.props.client.VtkGeometryDelivery.getArray);
+
+    const container = this.rootContainer;
+
+    // container.className = style.renderWindowContainer;
+    container.setAttribute('data-view-id', this.props.viewId);
+
+    // VTK renderWindow/renderer
+    const renderWindow = vtkSynchronizableRenderWindow.newInstance({ synchronizerContextName: SYNCHRONIZATION_CONTEXT_NAME });
+
+    // OpenGlRenderWindow
+    const openGlRenderWindow = vtkOpenGLRenderWindow.newInstance();
+    openGlRenderWindow.setContainer(container);
+    renderWindow.addView(openGlRenderWindow);
+
+    // Interactor
+    const interactor = vtkRenderWindowInteractor.newInstance();
+    interactor.setView(openGlRenderWindow);
+    interactor.initialize();
+    interactor.bindEvents(container);
+
+    this.props.client.VtkGeometryDelivery.onViewChange((data) => {
+      console.log('Received scene desciption:');
+      console.log(data);
+      renderWindow.synchronize(data[0]);
+    });
+
+    this.props.client.VtkGeometryDelivery.addViewObserver(this.props.viewId).then((successResult) => {
+      console.log(`Subscribe to view ${this.props.viewId} succeeded`);
+      console.log(successResult);
+    }, (failureResult) => {
+      console.log(`Failed to subscribe to view ${this.props.viewId}`);
+      console.log(failureResult);
+    });
+
+    // Handle window resize
+    function updateRenderWindowSize() {
+      const dims = container.getBoundingClientRect();
+      openGlRenderWindow.setSize(dims.width, dims.height);
+      renderWindow.render();
+    }
+
+    window.addEventListener('resize', updateRenderWindowSize);
+    updateRenderWindowSize();
+  },
+
+  componentWillUnmount() {
+    this.props.client.VtkGeometryDelivery.removeViewObserver(this.props.viewId).then((successResult) => {
+      console.log(`Unsubscribe from view ${this.props.viewId} succeeded`);
+      console.log(successResult);
+    }, (failureResult) => {
+      console.log(`Failed to unsubscribe from view ${this.props.viewId}`);
+      console.log(failureResult);
+    });
+
+    // Get our hands on the default synchronization context and clean it up
+    const synchCtx = vtkSynchronizableRenderWindow.getSynchronizerContext(SYNCHRONIZATION_CONTEXT_NAME);
+    synchCtx.emptyCachedInstances();
+    synchCtx.emptyCachedArrays();
+  },
```

Comment:
Probably yes props to control it.  I put it in so I wouldn't forget that we may not want all that context building over time.